pkg/build/admission/admission.go

```diff
@@ -0,0 +1,78 @@
+package admission
+
+import (
+	"io"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/admission"
+	kclient "github.com/GoogleCloudPlatform/kubernetes/pkg/client"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/runtime"
+
+	authorizationapi "github.com/openshift/origin/pkg/authorization/api"
+	buildapi "github.com/openshift/origin/pkg/build/api"
+	"github.com/openshift/origin/pkg/client"
+)
+
+type buildByStrategy struct {
+	*admission.Handler
+	client client.Interface
+}
+
+func init() {
+	admission.RegisterPlugin("BuildByStrategy", func(c kclient.Interface, config io.Reader) (admission.Interface, error) {
+		osclient := c.(client.Interface)
+		return NewBuildByStrategy(osclient), nil
+	})
+}
+
+// NewBuildByStrategy returns an admission control for builds that checks
+// on policy based on the build strategy type
+func NewBuildByStrategy(client client.Interface) admission.Interface {
+	return &buildByStrategy{
+		Handler: admission.NewHandler(admission.Create),
+		client:  client,
+	}
+}
+
+func (a *buildByStrategy) Admit(attr admission.Attributes) error {
+	if resource := attr.GetResource(); resource != "builds" {
```

Comment:
I think build configs are going to need to be gate too - just with a reject instead of mutate -> fail

> On Jun 12, 2015, at 8:46 PM, Jordan Liggitt <notifications@github.com> wrote:
> 
> In pkg/build/admission/admission.go:
> 
> > +		osclient := c.(client.Interface)
> > +		return NewBuildByStrategy(osclient), nil
> > +	})
> > +}
> > +
> > +// NewBuildByStrategy returns an admission control for builds that checks
> > +// on policy based on the build strategy type
> > +func NewBuildByStrategy(client client.Interface) admission.Interface {
> > +	return &buildByStrategy{
> > +		Handler: admission.NewHandler(admission.Create),
> > +		client:  client,
> > +	}
> > +}
> > +
> > +func (a *buildByStrategy) Admit(attr admission.Attributes) error {
> > +	if resource := attr.GetResource(); resource != "builds" {
> though it looks like buildconfigs/instantiate does the same thing, which would let you use a buildconfig to create a build bypassing admission
> 
> —
> Reply to this email directly or view it on GitHub.
>